From 9eb2b4dfc136da326e54081ae18cb4d648c6500d Mon Sep 17 00:00:00 2001
From: Greg Hudson <ghudson@mit.edu>
Date: Thu, 29 Nov 2012 01:58:13 -0500
Subject: [PATCH] Fix spin-loop bug in k5_sendto_kdc

In the second part of the first pass over the server list, we passed
the wrong list pointer to service_fds, causing it to see only a subset
of the server entries corresponding to sel_state.  This could cause
service_fds to spin if an event is reported on an fd not in the
subset.

(cherry picked from commit 2b06a22f7fd8ec01fb27a7335125290b8ceb6f18)

ticket: 7454
version_fixed: 1.10.4
status: resolved
---
 src/lib/krb5/os/sendto_kdc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/krb5/os/sendto_kdc.c b/src/lib/krb5/os/sendto_kdc.c
index 63dbcd8..22cd908 100644
--- a/src/lib/krb5/os/sendto_kdc.c
+++ b/src/lib/krb5/os/sendto_kdc.c
@@ -1287,7 +1287,7 @@ k5_sendto(krb5_context context, const krb5_data *message,
             continue;
         if (maybe_send(context, state, sel_state, callback_info))
             continue;
-        done = service_fds(context, sel_state, 1, state, seltemp, msg_handler,
+        done = service_fds(context, sel_state, 1, conns, seltemp, msg_handler,
                            msg_handler_data, &winner);
     }
 
-- 
2.9.2

